app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SuperscriptPlugin.java

```diff
@@ -0,0 +1,158 @@
+                return span;
+        return null;
+    }
+
+    private static SpannableBuilder.Span matchNonTextSpanAtBoundary(List<SpannableBuilder.Span> list, int value) {
+        for (var span : list)
+            if (span.what.getClass() != TextViewSpan.class && (span.end == value || span.start == value))
+                return span;
+        return null;
+    }
+
+    @Override
+    public void configureSpansFactory(@NonNull MarkwonSpansFactory.Builder builder) {
+        builder.setFactory(Superscript.class, (config, renderProps) -> new SuperscriptSpan());
+    }
+
+    @Override
+    public void configureVisitor(@NonNull MarkwonVisitor.Builder builder) {
+        builder.on(Superscript.class, new MarkwonVisitor.NodeVisitor<>() {
+            int depth = 0;
+
+            @Override
+            public void visit(@NonNull MarkwonVisitor visitor, @NonNull Superscript superscript) {
+                int start = visitor.length();
+
+                if (!superscript.isBracketed()) {
+                    superscriptOpeningList.add(new SuperscriptOpening(superscript, start));
+                    return;
+                }
+
+                depth++;
+                visitor.visitChildren(superscript);
+                depth--;
+                if (depth == 0) {
+                    int end = visitor.builder().length();
+                    var spannableStringBuilder = visitor.builder().spannableStringBuilder();
+                    var codeSpans = spannableStringBuilder.getSpans(start, end, CodeSpan.class);
+                    if (codeSpans.length > 0) {
+                        for (CodeSpan span : codeSpans) {
+                            if (spannableStringBuilder.getSpanEnd(span) <= end) {
+                                visitor.builder().setSpan(new SuperscriptSpan(), start, spannableStringBuilder.getSpanStart(span));
+                                visitor.builder().setSpan(new SuperscriptSpan(), spannableStringBuilder.getSpanEnd(span), end);
+                            }
+                        }
+                    } else {
+                        visitor.setSpansForNode(superscript, start);
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public void afterRender(@NonNull Node node, @NonNull MarkwonVisitor visitor) {
+        superscriptOpeningBracketStorage.clear();
+    }
+
+    @Override
+    public void configure(@NonNull Registry registry) {
+        registry.require(MarkwonInlineParserPlugin.class, plugin -> {
+                    plugin.factoryBuilder().addInlineProcessor(new SuperscriptOpeningInlineProcessor(superscriptOpeningBracketStorage));
+                    plugin.factoryBuilder().addInlineProcessor(new SuperscriptClosingInlineProcessor(superscriptOpeningBracketStorage));
+                }
+        );
+    }
+
+    @Override
+    public void afterSetText(@NonNull TextView textView) {
```

Comment:
Doesn't work with tables, probably because markdown is parsed only once, but `afterSetText` is called for every cell in the table

Reply:
Should be ok now

Reply:
Nope, you can test on
```
|a^superscript|b^another|
|---|---|
|one^more|final^one|
```

Positions stored in `superscriptOpeningList` appear to be correct because `TablePlugin` modifies `visitor.builder`, though I'm not 100% sure how it works. But you cannot rely on `superscriptOpeningList` anyways, for example `opening` and `nextOpening` can be from different table cells. I'd get rid of `superscriptOpeningList` and just get all `SuperscriptSpan`s from`textView`.

Reply:
By the way, just checked my example in current release and it is broken completely there. So this pr is already a huge improvement in terms of table compatibility!